themes/src/main/resources/theme/base/login/resources/js/passwordVisibility.js

```diff
@@ -0,0 +1,16 @@
+export default class PasswordVisibility {
+    static toggle(button, id) {
+        const passwordElement = document.getElementById(id);
+        if (passwordElement.type === "password") {
+            passwordElement.type = "text";
+            button.children.item(0).classList.replace("fa-eye", "fa-eye-slash");
+            button.setAttribute("aria-label", button.dataset.labelHide);
+        } else if(passwordElement.type === "text") {
+            passwordElement.type = "password";
+            button.children.item(0).classList.replace("fa-eye-slash", "fa-eye");
+            button.setAttribute("aria-label", button.dataset.labelShow);
+        }
+    }
+}
+
+window.PasswordVisibility = PasswordVisibility;
```

Comment:
Instead of exposing this as a global, I think it would be better if we found the elements on the page that are toggles by a special attribute. For example:

```html
<button
  type="button"
  aria-label="${msg('showPassword')}"
  data-password-toggle
  data-label-show="${msg('showPassword')}"
  data-label-hide="${msg('hidePassword')}"
></button>
```

These elements can then be retrieved in this file without leaking into the global scope:


```js
document.querySelectorAll('[data-password-toggle]');
```